src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -222,8 +222,7 @@ public void Start()
                 Debug.Assert(_delegate.Target is ThreadHelper);
                 var t = (ThreadHelper)_delegate.Target;
 
-                ExecutionContext? ec = ExecutionContext.Capture();
-                t.SetExecutionContextHelper(ec);
+                t.SetExecutionContextHelper(CurrentThread._executionContext);
```

Comment:
Using `Capture` prevents flow-suppressed ECs from flowing, reverting this part, doesn't make much of a difference
```suggestion
                ExecutionContext? ec = ExecutionContext.Capture();
                t.SetExecutionContextHelper(ec);
```